libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -410,6 +410,43 @@ bool ExpressionCompiler::visit(TupleExpression const& _tuple)
 bool ExpressionCompiler::visit(UnaryOperation const& _unaryOperation)
 {
 	CompilerContext::LocationSetter locationSetter(m_context, _unaryOperation);
+
+	if (FunctionDefinition const* function = _unaryOperation.annotation().userDefinedFunction)
+	{
+		FunctionType const* functionType = dynamic_cast<FunctionType const*>(
+			function->libraryFunction() ? function->typeViaContractName() : function->type());
+
+		solAssert(functionType);
+		functionType = dynamic_cast<FunctionType const&>(*functionType).asBoundFunction();
+		solAssert(functionType);
+		evmasm::AssemblyItem returnLabel = m_context.pushNewTag();
+		_unaryOperation.subExpression().accept(*this);
+		m_context << m_context.functionEntryLabel(*function).pushTag();
+
+		unsigned parameterSize =
+			CompilerUtils::sizeOnStack(functionType->parameterTypes()) +
+			functionType->selfType()->sizeOnStack();
+
+		if (m_context.runtimeContext())
```

Comment:
I don't think we need to do this here - I believe this is only required when you use `pushCombinedEntryLabel`.